D103-D106 QDYNAM instrumentation (ATMDYN.f part: ffde_in/ffde_q0/ffde_ain/ffde_out/ffde_fin inside QDYNAM; second USE TRACER_ADV for the AADVQ0 outputs). Helpers are in ATM_DRV_dynE.f.patch (units 1140-1149). Generated by diff -u against the PRISTINE ATMDYN.f (2026-10-05); local hunks.

--- ATMDYN.f.orig
+++ ATMDYN.f
@@ -3053,6 +3053,10 @@
       Use GEOM,       Only: AXYP
       USE TRACER_ADV, only:
      *    AADVQ,AADVQ0,sbf,sbm,sfbm,scf,scm,sfcm,ncyc,scf3d
+      USE TRACER_ADV, only:
+     *    ncycxy,nstepx,nstepz_extra,lminzij,lmaxzij,mw_extra,
+     *    pv_south,ni_checkfobs_y,ni_checkfobs_z,i_checkfobs_y,
+     *    i_checkfobs_z,do_z_extra
       USE DOMAIN_DECOMP_ATM, only: grid, getDomainBounds
       USE DOMAIN_DECOMP_1D, only : halo_update, south, north
       IMPLICIT NONE
@@ -3066,7 +3070,11 @@
 
       Do L=1,LM
          MMB(:,:,L) = MAOLD(L,:,:)*KG2MB*AXYP(:,:)  ;  EndDo
+      call ffde_in()
       Call AADVQ0  !  uses fluxes MUs,MVs,MWs from DYNAM
+      call ffde_q0(ncyc,do_z_extra,ncycxy,nstepx,nstepz_extra,
+     &     lminzij,lmaxzij,mw_extra,pv_south,ni_checkfobs_y,
+     &     ni_checkfobs_z,i_checkfobs_y,i_checkfobs_z)
 C****
 C**** convert from concentration to mass units
 C****
@@ -3076,10 +3084,12 @@
          Q(I,J,L) = Q(I,J,L)*MMB(I,J,L)
          QMOM(:,I,J,L) = QMOM(:,I,J,L)*MMB(I,J,L)
       enddo; enddo; enddo
+      call ffde_ain(Q,QMOM)
 C**** ADVECT
         sfbm = 0.; sbm = 0.; sbf = 0.
         sfcm = 0.; scm = 0.; scf = 0.
       CALL AADVQ (Q,QMOM, .TRUE. ,'q       ')
+      call ffde_out(Q,QMOM,sbf,sbm,sfbm,scf,scm,sfcm,scf3d)
         AGC(:,:,jl_totntlh) = AGC(:,:,jl_totntlh) + sbf(:,:)
         AGC(:,:,jl_zmfntlh) = AGC(:,:,jl_zmfntlh)
      &    + sbm(:,:)*sfbm(:,:)*byim
@@ -3096,6 +3106,7 @@
         Q(I,J,L) = Q(I,J,L)*byMMA
         QMOM(:,I,J,L) = QMOM(:,I,J,L)*byMMA
       enddo; enddo; enddo
+      call ffde_fin(Q,QMOM)
 
       RETURN
       END SUBROUTINE QDYNAM
